src/app/util/mock/MockNodeConfig.cpp

```diff
@@ -26,6 +26,62 @@
 
 namespace chip {
 namespace Test {
+namespace internal {
+uint16_t SizeForType(EmberAfAttributeType type)
+{
+    switch (type)
+    {
+    case ZCL_BOOLEAN_ATTRIBUTE_TYPE:
+    case ZCL_INT8S_ATTRIBUTE_TYPE:
+    case ZCL_INT8U_ATTRIBUTE_TYPE:
+    case ZCL_BITMAP8_ATTRIBUTE_TYPE:
+    case ZCL_ENUM8_ATTRIBUTE_TYPE:
+        return 1;
+    case ZCL_INT16S_ATTRIBUTE_TYPE:
+    case ZCL_INT16U_ATTRIBUTE_TYPE:
+    case ZCL_BITMAP16_ATTRIBUTE_TYPE:
+    case ZCL_ENUM16_ATTRIBUTE_TYPE:
+        return 2;
+    case ZCL_INT24S_ATTRIBUTE_TYPE:
+    case ZCL_INT24U_ATTRIBUTE_TYPE:
+        return 3;
+    case ZCL_SINGLE_ATTRIBUTE_TYPE:
+    case ZCL_INT32S_ATTRIBUTE_TYPE:
+    case ZCL_INT32U_ATTRIBUTE_TYPE:
+    case ZCL_BITMAP32_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_INT40S_ATTRIBUTE_TYPE:
+    case ZCL_INT40U_ATTRIBUTE_TYPE:
+        return 5;
+    case ZCL_INT48S_ATTRIBUTE_TYPE:
+    case ZCL_INT48U_ATTRIBUTE_TYPE:
+        return 6;
+    case ZCL_INT56S_ATTRIBUTE_TYPE:
+    case ZCL_INT56U_ATTRIBUTE_TYPE:
+        return 7;
+    case ZCL_DOUBLE_ATTRIBUTE_TYPE:
+    case ZCL_INT64S_ATTRIBUTE_TYPE:
+    case ZCL_INT64U_ATTRIBUTE_TYPE:
+    case ZCL_BITMAP64_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_CHAR_STRING_ATTRIBUTE_TYPE:
+    case ZCL_LONG_CHAR_STRING_ATTRIBUTE_TYPE:
+    case ZCL_OCTET_STRING_ATTRIBUTE_TYPE:
+    case ZCL_LONG_OCTET_STRING_ATTRIBUTE_TYPE:
+        return kDefaultStringSize;
+    case ZCL_ARRAY_ATTRIBUTE_TYPE:
+    case ZCL_STRUCT_ATTRIBUTE_TYPE:
+        // These items are using AAI and are acceptable for tests
+        return 0;
+    default:
+        // this type of attribute is not supported for tests, we cannot guess its size
+        //
+        // see attribute-type.h for a list of attributes
+        ChipLogError(Test, "Warning: size for attribute type 0x%x is not set", type) return 0;
```

Comment:
If this is used, could it lead to errors? If so, instead of logging, why not verify or die?

Reply:
I had VerifyOrDie first, however we do have a copy&paste of a lot of attribute types (all attribute types) in our unit test code, so for now I let it log so that tests pass without crashing on static init.

Making this VerifyOrDie requires me to do more yak shaving across tests that use mocks.